lib/async.js

```diff
@@ -0,0 +1,17 @@
+const RiffError = require('./riff-error');
+
+module.exports.guardWithTimeout = async (hookFn, timeoutInMs) => {
```

Comment:
since the function always returns a promise, making it also async add overhead and no value.

```suggestion
module.exports.guardWithTimeout = (hookFn, timeoutInMs) => {
```